odoo/addons/base/models/ir_actions.py

```diff
@@ -420,6 +423,31 @@ def _check_recursion(self):
         if not self._check_m2m_recursion('child_ids'):
             raise ValidationError(_('Recursion found in child server actions'))
 
+    def _get_runner(self):
+        multi = True
+        t = type(self)
+        fn = getattr(t, f'_run_action_{self.action_type}_multi', None)\
+          or getattr(t, f'run_action_{self.action_type}_multi', None)
+        if not fn:
+            multi = False
+            fn = getattr(t, f'_run_action_{self.action_type}', None)\
+              or getattr(t, f'run_action_{self.action_type}', None)
+        if fn.startswith('run_action_'):
+            fn = functools.partial(fn, self)
+        return fn, multi
+
+    def _register_hook(self):
+        super()._register_hook()
+
+        for cls in type(self).mro():
+            for symbol, val in vars(cls).items():
+                m = RUNNER_PATTERN.match(symbol) or {'private': True}
+                if not m['private']:
```

Comment:
The inner loop can be simpler, as `val` is unused.

The definition of `m` is very clever, but *way too clever* to be understood at first sight.
It could be as simple and stupid as:
```python
if PUBLIC_RUNNER_PATTERN.match(symbol):
```
where
```python
PUBLIC_RUNNER_PATTERN = re.compile(r"run_action_\w+(_multi)?$")
```

Reply:
Oh yeah, basically the same reason as to why the regex has more groups than it needs, originally this would create a map of action types to callbacks (with a multi flag) right after the registry was loaded so there was no lookup afterwards, but it turned out to kinda suck, and I just removed the "create a callbacks map" without stripping out the rest.

Reply:
I can just make this into `symbol.startswith('run_action_')` at this point.